src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs

```diff
@@ -218,32 +214,49 @@ private static void CheckMercurialIni()
 					// should have already set this correctly, so that the check above would pass.
 					// review: Is there a better way to do this, so that this test only code is not
 					// included in the main code? CP 2012-04
-					SetExtensions(doc, extensions);
+					SetExtensions(doc, HgExtensions);
 					try
 					{
 						doc.Save();
 					}
-					// ReSharper disable EmptyGeneralCatchClause
-					catch (Exception)
+						// ReSharper disable EmptyGeneralCatchClause
+					catch(Exception)
 					{
 					}
 					// ReSharper restore EmptyGeneralCatchClause
 					doc = GetMercurialConfigInMercurialFolder();
-					if (!CheckExtensions(doc, extensions))
+					if(!CheckExtensions(doc, extensions))
 					{
 						throw new ApplicationException(
 							"The mercurial.ini file shipped with this application does not have the fixutf8 extension enabled."
-						);
+							);
 					}
 				}
 				_alreadyCheckedMercurialIni = true;
 			}
-			catch (Exception error)
+			catch(Exception error)
 			{
 				throw new ApplicationException(string.Format("Failed to set up extensions: {0}", error.Message));
 			}
 		}
 
+		private void EnsureChorusMergeAddedToHgrc()
+		{
+			var mergetoolname = "chorusmerge";
+			var doc = GetMercurialConfigForRepository();
+			var chorusMergeLoc = SurroundWithQuotes(ExecutionEnvironment.ChorusMergeFilePath());
+			var uiSection = doc.Sections.GetOrCreate("ui");
+
+			if(uiSection.GetValue("merge") != mergetoolname)
+			{
+				uiSection.Set("merge", mergetoolname);
+				var mergeToolsSection = doc.Sections.GetOrCreate("merge-tools");
+				mergeToolsSection.Set(String.Format("{0}.premerge", mergetoolname), "False");
```

Comment:
If I'm understanding at all, this is the core of how we fixed a problem that required weeks of effort. I think it deserves a comment linking it to some of the issues and explaining why premerge is a disaster for Chorus.